### PR TITLE
Teamnado 4186 to merge into main 

### PR DESCRIPTION
**Pull request by BrooklynLCodes**
# Description

Associated Jira ticket: # (issue)

Please include a summary of the change, what this fixes/creates/improves.

# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description
- [x] Screenshots before and after the change are added
- [x] Tests for the changes have been added
- [x] README.md is updated if necessary
- [x] Needs additional dependent work

### CODE REVIEW

**Comment by Fewwy** [comment]
@BrooklynLCodes Sorry for the ping. But please, add a Jira ticket link and description so we could review the PR

**Comment by BrooklynLCodes** [comment]
> @BrooklynLCodes Sorry for the ping. But please, add a Jira ticket link and description so we could review the PR

My apologies here is the Jira Link[ ](https://issues.redhat.com/browse/TEAMNADO-4186)


https://issues.redhat.com/browse/TEAMNADO-4186

File: src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js
Comment (cowanrc) [inline src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js]: I don't think this search filter is working as intended. I would maybe check how it's done in subscription inventory, but the search wasn't filtering correctly when using the edit modal.  

here's how we do it with product table, but the logic will need to change a little:

  const filterDataBySearchTerm = (data: Product[], searchValue: string): Product[] => {
    return data.filter((entry: Product) => {
      const searchTerm = searchValue.toLowerCase().trim();
      const name = (entry.name || '').toLowerCase();
      const productLine = (entry.productLine || '').toLowerCase();
      const sku = (entry.sku || '').toLowerCase();
      return (
        name.includes(searchTerm) || productLine.includes(searchTerm) || sku.includes(searchTerm)
      );
    });
  };

File: src/Components/Modals/EditAdditionalRepositoriesModal.js
Comment (cowanrc) [inline src/Components/Modals/EditAdditionalRepositoriesModal.js]: is this save changes button supposed to permanently be disabled?

File: src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js
Comment (cowanrc) [inline src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js:214]: The sorting seems to be breaking the pagination. Again, I'd look how subscription inventory handles the sorting but there seems to be some sort of functions being used towrds the end that keeps pagination working

 const sortedProducts = sortProducts(data, activeSortIndex);
  const searchedProducts = filterDataBySearchTerm(sortedProducts, searchValue);
  const paginatedProducts = getPage(searchedProducts);
  
  These will need to be changed to fit the repos, but just pointing it out

**Review by cowanrc — CHANGES_REQUESTED** [review]

File: src/Components/Modals/EditAdditionalRepositoriesModal.js
Comment (BrooklynLCodes) [reply src/Components/Modals/EditAdditionalRepositoriesModal.js]: I had it disabled as it was in the wireframe sketch , but i took that attribute out for future edits !

Comment (jmbuscemi) [reply src/Components/Modals/EditAdditionalRepositoriesModal.js]: Just to clarify, for this story we aren't really doing anything with this modal except showing the data and sorting/searching.  The functionality behind that Save button will come in a later story.

At this point it doesn't matter if Save is enabled or disabled, so I'll let you all decide how you want it.  I just wanted to provide the context.

File: src/Components/Modals/EditAdditionalRepositoriesModal.js
Comment (seodma) [inline src/Components/Modals/EditAdditionalRepositoriesModal.js]: you're using isLoading and error props in EditAdditionalRepositoriesTable, but never pass them

File: src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js
Comment (seodma) [inline src/Components/EditAdditionalRepositoriesTable/EditAdditionalRepositoriesTable.js:384]: Since you're passing repositories as an prop -  I think it would be better to extract Loading/Unavailable logic to parent component. Or alternatively pass activation_key as prop and fetch repositories inside this component, that way you will have access to isLoading and error state from react-query

**Comment by Fewwy** [comment]
@BrooklynLCodes Incorrect margin for the "Edit additional repositories" button
image

**Comment by Fewwy** [comment]
@BrooklynLCodes With the last updates, the incorrect margin is fixed. I still don't have a way to see the table values and check the sorting. If the jenkins errors are fixed and I can test the table it should be Ok

**Comment by BrooklynLCodes** [comment]
> @BrooklynLCodes With the last updates, the incorrect margin is fixed. I still don't have a way to see the table values and check the sorting. If the jenkins errors are fixed and I can test the table it should be Ok

Ok I have fixed the Jenkins issue.
So the table for this A/C it is  opened with the click of the  "Edit additional repositories" button that opens the modal on the "additional repos card". The modal represents the table with the "available repos" pulled from the v2/available_repos endpoint and all the criteria for the A/C. You should have access to this modal + table with the login credentials provided offline.

**Comment by Fewwy** [comment]
@jiridostal pinging you so you will be informed about this and tell me if I have a wrong understanding of the issue.
I tested this PR locally again and talked with Brooklyn, everything that is "ready" works fine. Overall it's a great job. Thanks @BrooklynLCodes 

But we have an issue with merging this into the master/main branch. Brooklyn said that there are parts like the "Save changes" button that doesn't have the backend ready. That means if we merge it right now and THEN we'll have other PR that should go into production unfinished work from this PR could get into production. 
There are 2 solutions for this:
Feature flags or wait until this PR is 100% ready to be merged.

**Comment by seodma** [comment]
@Fewwy @jiridostal I'm going to mention that Activation Key details page is already behind a feature flag
https://github.com/RedHatInsights/sed-frontend/blob/a4049f9eda08fac71316711d0c5a476a7b79a992/src/Components/ActivationKeysTable/ActivationKeysTable.js#L46

**Comment by Fewwy** [comment]
@seodma If it's behind a feature flag and they are set up correctly, we need to hide it for the time being and as soon we confirm that it is indeed hidden - we can merge it

**Review by Fewwy — APPROVED** [review]

**Pull request merged**
